### ISSUE TITLE
preview: misleading error when cluster is unreachable (chart requires kubeVersion: >= x.y.z which is incompatible with Kubernetes va.b.c)

### ISSUE THREAD

**Issue by awoimbee** [pulumi/pulumi-kubernetes#2374]
### What happened?

Hi,

TL;DR: cluster is unreachable but only error printed during preview is `chart requires kubeVersion: >= x.y.z`.

I did deployed a fresh stack of a pulumi deployment and got hit by `chart requires kubeVersion: >= x.y.z` when I used pulumi up (during the preview stage).

I then used the "I'm a pro" shortcut: `pulumi up --yes --skip-preview --non-interactive`, which returned the correct error: `configured Kubernetes cluster is unreachable: unable to load Kubernetes client configuration from kubeconfig file. [...]`.
Indeed, my kubeconfig was outdated !

### Expected Behavior

The correct error is printed somewhere (`unable to load Kubernetes client configuration [...]`).

The best would be for it to be included in an error message. But if not possible, it should at least be in a warning (`Kubernetes cluster is unreachable (unable to load [...]), preview might fail`).

### Steps to reproduce

Sorry, don't have time for a simple copy-paste MRE right now:

- create pulumi-kubernetes project
- use a helm chart the requires a recent k8s version (higher than [the helm default](https://github.com/helm/helm/blob/14047fed6f2f63d3f66303433f5807cec9560aff/pkg/chartutil/capabilities.go#L34))
- make the k8s provider point to a non-existant kubectl context (I have not tried other failure modes)
- kaboom

### Output of `pulumi about`

Wow, `pulumi about` leaks a lot of info !
```
CLI
Version      3.66.0
Go Version   go1.20.3
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host
OS       arch
Version  22.1.0
Arch     x86_64

This project is written in nodejs: executable='/usr/bin/node' version='v19.9.0'

Current Stack: [redacted]

TYPE                                                        URN
[redacted]

Found no pending operations associated with [redacted]

Backend
Name           [redacted]
URL            s3://[redacted]
User           [redacted]
Organizations
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by thomas11** [comment]
Thank you for reporting this, @awoimbee! We'll queue it up to look into.

**Comment by BrennerSpear** [comment]
following up here since we also ran into this problem this week

**Comment by mjeffryes** [comment]
We intentionally don't treat not being able to hit the cluster as a blocking error during preview so it doesn't prevent previews for programs that create the cluster and then deploy resources into it. But we could probably be more thoughtful about how we handle other errors during the preview, such as flagging that we couldn't reach the cluster and always mentioning that in the error message.
